### ISSUE TITLE
pdf.worker.js file not detected correctly for urls ending with a query parameter

### ISSUE THREAD

**Issue by ngzhian** [mozilla/pdf.js#5710]
I'm using [tornado](http://www.tornadoweb.org/en/stable/web.html#tornado.web.RequestHandler.static_url) as my web server, and the built in static file server appends a query string to the end of static files. e.g. `pdf.js` ends up being like this:

```
/static/js/lib/pdf.js?v=e0130215dbf4b94e43fc2039af2ab91e
```

In that case this piece of code right at the bottom of `pdf.js` fails:

``` js
  PDFJS.workerSrc = (function () {
    'use strict';
    var scriptTagContainer = document.body ||
                             document.getElementsByTagName('head')[0];
    var pdfjsSrc = scriptTagContainer.lastChild.src;
    return pdfjsSrc && pdfjsSrc.replace(/\.js$/i, '.worker.js');
  })();
```

the regex fails because it didn't catch the `?v=e0130215dbf4b94e43fc2039af2ab91e` at the end.
Not sure if we want to support this, but there're two workarounds:
1. match `pdf.js` and replace with `pdf.worker.js`
2. make regex more generic, something along the lines of `/\.js\?+.+$/`

be happy to submit a patch if this is accepted, thank you!

**Comment by ciosbel** [comment]
Just to bump the original issue, since still applies to the latest stable release and it took me quite a bit to figure it out.
PDF.js stopped working (the getDocument() promise never resolves with any error/message) from a day to another by simply appending a query string to the served js file.

Thank you.

**Comment by yurydelendik** [comment]
Fixed by #8309
